Core/Src/main.c

```diff
@@ -87,16 +89,39 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  uint8_t transmit_data[3] = {0x01, 0x80, 0x00};
+  uint8_t receive_data[3] = {0x0, 0x0, 0x0};
+  uint16_t adc_value;
+  float adc_norm;
+
+  // Pull CS line high to end transmission.
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+  __HAL_TIM_SET_COMPARE(&htim1, TIM_CHANNEL_1, 0.05 * 64000);
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
     /* USER CODE END WHILE */
-
+	  // Pull CS line low to start transmission.
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+	  // ADC requires full duplex mode SPI communication.
+	  HAL_SPI_TransmitReceive(&hspi1, transmit_data, receive_data, sizeof(transmit_data), HAL_MAX_DELAY);
+	  // Pull CS line high to end transmission.
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+	  // Concatenate 10 bits of raw data from ADC stored in receive_data. Normalize ADC value to be in between 0.05 to 1.
+	  adc_value = ((receive_data[1] & 0x03) << 8) | receive_data[2];
+	  adc_norm = ((adc_value / 1023)*0.05) + 0.05;
+
+	  // Compares counter value to register.
+	  __HAL_TIM_SET_COMPARE(&htim1, TIM_CHANNEL_1, adc_norm);
```

Comment:
`__HAL_TIM_SET_COMPARE` function takes the number ticks as the input parameter, it doesn't work automatically by telling it the duty cycle unfortunately.